server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/analysis/Project.java

```diff
@@ -77,15 +86,13 @@ public String toString() {
     sb.append("uuid='").append(uuid).append('\'');
     sb.append(", key='").append(key).append('\'');
     sb.append(", name='").append(name).append('\'');
+    sb.append(", description=").append(toString(this.description));
     sb.append('}');
     return sb.toString();
   }
 
-  public static Project copyOf(Component component) {
-    return new Project(component.getUuid(), component.getKey(), component.getName());
+  private static String toString(@Nullable String s) {
+    return s == null ? null : '\'' + s + '\'';
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Add parentheses to make the operator precedence explicit. ([squid:S864](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS864))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fsl%2FSONAR-10261%2Fbranch_analysis_does_not_update_name_description_nor_links&issues=AWH_t2Bo9MctqhxriUSu&open=AWH_t2Bo9MctqhxriUSu)

[ISSUE_KEY]: AWH_t2Bo9MctqhxriUSu